### PR TITLE
Adding frameless unit test cases to app capability

### PR DESCRIPTION
**Pull request by shrshindeMSFT**
- Adding missing frameless unit test cases to app capability.

### CODE REVIEW

File: packages/teams-js/test/public/app.spec.ts [hunk 1]
```diff
@@ -11,585 +11,1213 @@ import {
...
-      utils.respondToMessage(initMessage, FrameContexts.content, HostClientType.web, 'nonJSONStr', '1.6.0');
-      await initPromise;
+        const initMessage = utils.findMessageByFunc('initialize');
+        utils.respondToMessage(initMessage, FrameContexts.content, HostClientType.web, '1.6.0', 'nonJSONStr');
+        await initPromise;
 
-      expect(runtime).toEqual(teamsRuntimeConfig);
+        expect(runtime).toEqual(teamsRuntimeConfig);
+      });
     });
```
Comment (erinha) [inline packages/teams-js/test/public/app.spec.ts:230]: These are great additions! Unless I'm missing something, I think there are some more test cases that should be added to validate `app.initialize`:

- validating GlobalVars.hostClientType is set correctly
- validating the various capabilities `.initialize` functions are called (e.g., `authentication.initialize()`, `menus.initialize()`, etc.)
- validating GlobalVars.additionalValidOrigins is set correctly

File: packages/teams-js/test/public/app.spec.ts [hunk 2]
```diff
@@ -11,585 +11,1213 @@ import {
...
+          expect(message.args.length).toBe(2);
+          expect(message.args[0]).toEqual(app.ExpectedFailureReason.PermissionError);
+          expect(message.args[1]).toEqual('Failed message');
+        });
+      });
+    });
+
+    describe('Testing app.registerOnThemeChangeHandler function', () => {
+      it('app.registerOnThemeChangeHandler should not allow calls before initialization', () => {
+        expect(() => app.registerOnThemeChangeHandler(() => {})).toThrowError(
```
Comment (erinha) [inline packages/teams-js/test/public/app.spec.ts:1120]: Can we define an emptyCallback and use it in the two places we're using this?

Comment (erinha) [reply packages/teams-js/test/public/app.spec.ts:1120]: And then we can add the `// eslint-disable-next-line @typescript-eslint/no-empty-function` above the emptyCallback declaration

File: packages/teams-js/test/public/app.spec.ts [hunk 3]
```diff
@@ -11,585 +11,1213 @@ import {
...
+          data: {
+            id: initMessage.id,
+            args: [FrameContexts.content, HostClientType.web, '1.6.0', 'nonJSONStr'],
+          },
+        } as DOMMessageEvent);
+        await initPromise;
+
+        expect(runtime).toEqual(teamsRuntimeConfig);
+      });
     });
```
Comment (erinha) [inline packages/teams-js/test/public/app.spec.ts:837]: I think the same comments in the framed tests apply to the frameless ones for `app.initialize` test cases.

**Review by erinha — CHANGES_REQUESTED** [review]
Thanks for adding and organizing these tests! A couple comments.

File: packages/teams-js/test/public/app.spec.ts [hunk 4]
```diff
@@ -11,585 +11,1213 @@ import {
...
+
+          expect(actualContext).toEqual(expectedContext);
+          expect(actualContext.page.frameContext).toBe(context);
+          expect(actualContext.meeting.id).toBe('dummyMeetingId');
+        });
+      });
+    });
+
+    describe('Testing app.notifyAppLoaded function', () => {
+      it('app.notifyAppLoaded should not allow calls before initialization', () => {
```
Comment (erinha) [inline packages/teams-js/test/public/app.spec.ts:1044]: Should we both with having this test twice? I don't think there is anything different in the setup between framed and frameless, unless I'm missing something?

Comment (erinha) [reply packages/teams-js/test/public/app.spec.ts:1044]: Same with `app.notifySuccess` and `app.notifyFailure`

Comment (erinha) [reply packages/teams-js/test/public/app.spec.ts:1044]: And `app.registerOnThemeChangeHandler`

Comment (erinha) [reply packages/teams-js/test/public/app.spec.ts:1240]: Ah it turns out I'm wrong. There is a difference in setup between framed and frameless and that is the important part of these tests.

**Review by erinha — APPROVED** [review]

File: packages/teams-js/test/public/app.spec.ts [hunk 5]
```diff
@@ -11,585 +13,1408 @@ import {
...
+import { FramelessPostMocks } from '../framelessPostMocks';
 import { Utils } from '../utils';
 
-describe('AppSDK-app', () => {
-  // Use to send a mock message from the app.
-  const utils = new Utils();
-
+describe('Testing app capability', () => {
   const mockErrorMessage = 'Something went wrong...';
+  describe('Framed - Testing app capbility', () => {
```
Comment (TrevorJoelHarris) [inline packages/teams-js/test/public/app.spec.ts:22]: "capability"

**Review by TrevorJoelHarris — DISMISSED** [review]

File: packages/teams-js/test/public/app.spec.ts (hunk 5 above)
Comment (erinha) [reply packages/teams-js/test/public/app.spec.ts:22]: Thanks!

File: packages/teams-js/test/public/app.spec.ts [hunk 6]
```diff
@@ -11,585 +13,1408 @@ import {
...
+      });
     });
-  });
 
-  describe('Testing app.getFrameContext function', () => {
-    Object.values(FrameContexts).forEach(context => {
-      it(`app.getFrameContext should return ${context} context`, async () => {
-        await utils.initializeWithContext(context);
-        expect(app.getFrameContext()).toBe(context);
+    describe('Testing app.initialize function', () => {
```
[the remark this answers is not in the document]
Comment (erinha) [reply packages/teams-js/test/public/app.spec.ts:65]: Love it

File: packages/teams-js/test/public/app.spec.ts [hunk 7]
```diff
@@ -11,585 +13,1408 @@ import {
...
+        expect(initMessage.timestamp).not.toBeNull();
       });
-    });
-  });
 
-  describe('Testing app.initialize function', () => {
-    it('app.initialize should successfully initialize', () => {
-      app.initialize();
+      it('app.initialize should allow multiple initialize calls', () => {
+        for (let i = 0; i < 100; i++) {
```
[the remark this answers is not in the document]
Comment (erinha) [reply packages/teams-js/test/public/app.spec.ts:82]: It came from v1 tests. Happy to make it 2.

**Review by erinha — APPROVED** [review]

**Review by TrevorJoelHarris — APPROVED** [review]

**Pull request merged**